### PR TITLE
ci: improve integration workflow

### PR DESCRIPTION
**Pull request by abn**

### CODE REVIEW

**Comment by abn** [comment]
@sdispater @frostming can I get workflow approval please?

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -4,27 +4,99 @@ on:
...
+
+      - name: Ensure cache is healthy
+        if: steps.cache.outputs.cache-hit == 'true'
+        run: timeout 10s poetry run pip --version || rm -rf .venv
+
+      - name: Patch tomlkit dependency
+        if: matrix.project != 'poetry-core'
+        run: |
+          # we explicitly remove to mitigate a bug in poetry 1.1.13, can be removed with 1.2.0
+          poetry run pip uninstall tomlkit
```
Comment (frostming) [inline .github/workflows/integration.yml:86]: Don't we need `-y` here?

Comment (abn) [reply .github/workflows/integration.yml:86]: Good catch. Yes.

Comment (abn) [inline .github/workflows/integration.yml:86]:
```suggestion
          poetry run pip uninstall -y tomlkit
```

**Review by frostming — APPROVED** [review]

**Pull request merged**